File: youpinhui-manager-web/src/main/webapp/js/service/contentService.js
//Sevice
app.service('contentService',function($http){
	    	
	 //test case , get all the data return a Json
	this.findAll=function(){
		return $http.get('../content/findAll.do');		
	}
	//pagination
	this.findPage=function(page,rows){
		return $http.get('../content/findPage.do?page='+page+'&rows='+rows);
	}
	//find one object
	this.findOne=function(id){
		return $http.get('../content/findOne.do?id='+id);
	}
	//add 
	this.add=function(entity){
		return  $http.post('../content/add.do',entity );
	}
	//update 
	this.update=function(entity){
		return  $http.post('../content/update.do',entity );
	}
	//batch delete
	this.dele=function(ids){
		return $http.get('../content/delete.do?ids='+ids);
	}
	//search
	this.search=function(page,rows,searchEntity){
		return $http.post('../content/search.do?page='+page+"&rows="+rows, searchEntity);
	}    	
});

File: youpinhui-content-interface/src/main/java/com/youpinhui/content/service/ContentService.java
package com.youpinhui.content.service;

import java.util.List;
import com.youpinhui.pojo.TbContent;
import com.youpinhui.entity.PageResult;
import com.youpinhui.pojo.TbContentCategory;

/**
 * provider interface
 * @author Administrator
 *
 */
public interface ContentService {

	/**
	 * find all
	 * @return
	 */
	public List<TbContent> findAll();
	
	
	/**
	 * query according pagination 
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * Add
	*/
	public void add(TbContent content);
	
	
	/**
	 * update
	 */
	public void update(TbContent content);
	

	/**
	 * get one by id
	 * @param id
	 * @return
	 */
	public TbContent findOne(Long id);
	
	
	/**
	 * batch delete
	 * @param ids
	 */
	public void delete(Long [] ids);

	/**
	 * fuzzy search 
	 * if the object  is not null , get the parameter from it as the search condition.
	 * @param pageNum current page number
	 * @param pageSize size of the page to display
	 * @return
	 */
	public PageResult findPage(TbContent content, int pageNum,int pageSize);
	
	
	public List<TbContent> findByCategoryId(Long categoryId);
}
